monitoring/bin/create_elasticsearch_datasource_tenant.sh

```diff
@@ -0,0 +1,167 @@
+POS_PARMS=""
+
+# Setting passed in variables:
+while (( "$#" )); do
+  case "$1" in
+    -ns|--namespace)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenantNS=$2
+        shift 2
+      else
+        log_error "A value for parameter [NAMESPACE] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -t|--tenant)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenant=$2
+        shift 2
+      else
+        log_error "A value for parameter [TENANT] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -u|--user)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        userID=$2
+        shift 2
+      else
+        log_error "A value for parameter [USER] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -p|--password)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        passwd=$2
+        shift 2
+      else
+        log_error "A value for parameter [PASSWORD] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -h|--help)
+      show_usage
+      exit
+      ;;
+    -*|--*=) # unsupported flags
+      log_error "Unsupported flag $1" >&2
+      show_usage
+      exit 1
+      ;;
+    *) # preserve positional arguments
+      POS_PARMS="$POS_PARMS $1"
+      shift
+      ;;
+  esac
+done
+
+# Check to make sure all parameters have been provided
+if [ -z "$tenantNS" ]; then
+  log_error "A value for parameter [NAMESPACE] was not been provided."
+  exit 2
+fi
+
+if [ -z "$tenant" ]; then
+  log_error "A value for parameter [TENANT] was not been provided."
+  exit 2
+fi
+
+if [ -z "$userID" ]; then
+  log_error "A value for parameter [USER] was not been provided."
+  exit 2
+fi
+
+if [ -z "$passwd" ]; then
+  log_error "A value for parameter [PASSWORD] was not been provided."
+  exit 2
+fi
+
+# Convert namespace and tenant to all lower-case
+tenantNS=$(echo "$tenantNS"| tr '[:upper:]' '[:lower:]')
+tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
+
+# Check to see if monitoring deployment could be found for tenant in provided namespace
+log_info "Checking the $tenantNS namespace for monitoring deployment for the $tenant tenant ..."
+if [[ $(kubectl get pods -n $tenantNS -l app.kubernetes.io/instance=v4m-grafana-$tenant -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+  log_error "No monitoring components were found for $tenant in the $tenantNS namespace."
+  log_error "Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+  exit 1
+else
+  log_debug "Monitoring deployment was found for the $tenant tenant in the $tenantNS namespace.  Continuing."
+fi
+
+# Check to see if tenant has been deployed in logging.
+log_info "Checking that $tenant has been onboarded to Elasticsearch in the $LOG_NS namespace ..."
+if [[ $(kibana_tenant_exists $tenant) == 1 ]]; then
```

Comment:
You can simplify this syntax to:
`if ! kibana_tenant_exists "$ktenant"; then`

More importantly, I don't see a call to the get_sec_api function; this function must be called to establish the connection to Elasticsearch.  Is it being called somewhere else?